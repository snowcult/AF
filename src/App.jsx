import CardList from "./CardList.jsx";
import SearchBox from "./SearchBox.jsx";
import { Component } from "react";

class App extends Component {
  constructor() {
    super();
    this.state = {
      robots: [],
      searchField: "",
    };
  }
  async componentDidMount() {
    try {
      const res = await fetch("https://jsonplaceholder.typicode.com/users");
      const data = await res.json();
      this.setState({ robots: [...data] });
    } catch (e) {
      console.error(e.message);
    }
  }

  onSearchChange = (e) => {
    this.setState({ searchField: e.target.value.toLowerCase() });
  };
  render() {
    const filteredRobots = this.state.robots.filter((robot) => {
      return robot.name.toLowerCase().includes(this.state.searchField);
    });
    if (this.state.robots.length === 0) {
      return (
        <div className="tc">
          <h2 className="red">Loading...</h2>;
        </div>
      );
    } else {
      return (
        <div className="tc">
          <h1 className="tc f-5 white ">AiFriends</h1>
          <SearchBox searchChange={this.onSearchChange} />
          <CardList robots={filteredRobots} />
        </div>
      );
    }
  }
}
export default App;
