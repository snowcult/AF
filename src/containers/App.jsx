import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { useEffect } from 'react';
import Scroll from '../components/Scroll.jsx';
import ErrorBoundary from '../components/ErrorBoundary.jsx';
import { requestRobots, setSearchField } from '../actions/actions.js';
import Header from '../components/Header.jsx';
import MainPage from '../components/pages/Mainpage.jsx';

const mapStateToProps = state => {
  return {
    searchField: state.searchRobots.searchField,
    robots: state.requestRobots.robots,
    error: state.requestRobots.error,
    isPending: state.requestRobots.isPending,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onSearchChange: event => {
      return dispatch(setSearchField(event.target.value));
    },
    onRequestRobots: () => dispatch(requestRobots()),
  };
};

function App(props) {
  return <MainPage {...props} />;
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
