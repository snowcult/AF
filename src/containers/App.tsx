import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { requestRobots, setSearchField } from '../actions/actions.js';
import MainPage from '../components/pages/Mainpage.jsx';
import React from 'react';

interface IState {
  searchRobots: { searchField: string };
  requestRobots: { robots: string[]; error: string; isPending: boolean };
}
type dispatchs = {
  onRequestRobots: () => void;
};

const mapStateToProps = (state: IState) => {
  return {
    searchField: state.searchRobots.searchField,
    robots: state.requestRobots.robots,
    error: state.requestRobots.error,
    isPending: state.requestRobots.isPending,
  };
};

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => {
      return dispatch(setSearchField(event.target.value));
    },
    onRequestRobots: () => dispatch(requestRobots()),
  };
};

function App(props: Element) {
  return <MainPage {...props} />;
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
