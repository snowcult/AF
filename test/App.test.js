import { shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import App from 'src/containers/App.tsx';

const mockStore = configureStore([]);
const initialState = { robots: [], searchField: '' };
const store = mockStore(initialState);

describe('App component', function () {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <App />
      </Provider>
    );
  });
  it('expect to render App comp', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
