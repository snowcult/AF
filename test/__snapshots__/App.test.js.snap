// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component expect to render App comp 1`] = `
<ContextProvider
  value={
    {
      "getServerState": undefined,
      "store": {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": {
        "addNestedSub": [Function],
        "getListeners": [Function],
        "handleChangeWrapper": [Function],
        "isSubscribed": [Function],
        "notifyNestedSubs": [Function],
        "trySubscribe": [Function],
        "tryUnsubscribe": [Function],
      },
    }
  }
>
  <Connect(App) />
</ContextProvider>
`;
